src/main/java/com/crud/tasks/controller/TaskController.java

```diff
@@ -20,37 +20,87 @@
     @Autowired
     private TaskMapper taskMapper;
 
-    @RequestMapping(method = RequestMethod.GET)
+    // alias '/all' is consistent with convention applied to deleteAllTasks()
+    @RequestMapping(method = RequestMethod.GET, value = {"", "/all"})
     public List<TaskDto> getTasks() {
         return taskMapper.mapToTaskDtoList(service.getAllTasks());
     }
 
     @RequestMapping(method = RequestMethod.GET, value = "/{taskId}")
     public ResponseEntity<TaskDto> getTask(@PathVariable("taskId") Long taskId) {
-        validateId(taskId);
+        validateLegalId(taskId);
         return new ResponseEntity<>(
             taskMapper.mapToTaskDto(service.getTask(taskId).orElseThrow(NoSuchTaskException::new)),
             HttpStatus.OK);
     }
 
     @RequestMapping(method = RequestMethod.DELETE, value = "/{taskId}")
-    public ResponseEntity deleteTask(@PathVariable("taskId") Long taskId) {
-        throw new UnsupportedOperationException();
+    public ResponseEntity<TaskDto> deleteTask(@PathVariable("taskId") Long taskId) {
+        validateExistingId(taskId);
+        service.deleteTask(taskId);
+        return new ResponseEntity<>(HttpStatus.NO_CONTENT);
+    }
+
+    // supplementary '/all' is for security reason
+    @RequestMapping(method = RequestMethod.DELETE, value = "/all")
+    public ResponseEntity<TaskDto> deleteAllTasks() {
```

Comment:
Instead of appending "all", you can add restriction that this operation will be possible only if some additional header is provided, for example: "confirm_delete_all=true". Thanks yo that your api will be more RESTful